src/Cli.Tests/ConfigGeneratorTests.cs

```diff
@@ -0,0 +1,103 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Cli.Tests;
+
+/// <summary>
+/// Config Generation Tests for CLI.
+/// </summary>
+[TestClass]
+public class ConfigGeneratorTests
+{
+    /// <summary>
+    /// Setup the logger for CLI
+    /// </summary>
+    [ClassInitialize]
+    public static void Setup(TestContext context)
+    {
+        TestHelper.SetupTestLoggerForCLI();
+    }
+
+    /// <summary>
+    /// Tests that user provided config file is successfully generated when that file is not already present.
+    /// </summary>
+    [DataTestMethod]
+    [DataRow(true, false, DisplayName = "Failed to generate config file when user provided config file is present.")]
+    [DataRow(false, true, DisplayName = "Successfully generated config file when user provided config file is not present.")]
+    public void TryGenerateConfig_WithUserProvidedConfig(
+        bool isConfigFilePresentAlready,
+        bool isConfigGenerationSuccessful)
+    {
+        HandleConfigFileCreationAndDeletion(TEST_RUNTIME_CONFIG_FILE, isConfigFilePresentAlready);
+        Assert.AreEqual(isConfigFilePresentAlready, File.Exists(TEST_RUNTIME_CONFIG_FILE));
+
+        InitOptions options = CreateBasicInitOptionsForMsSqlWithConfig(config: TEST_RUNTIME_CONFIG_FILE);
+
+        Assert.AreEqual(isConfigGenerationSuccessful, ConfigGenerator.TryGenerateConfig(options));
+
+        if (!isConfigFilePresentAlready)
+        {
+            Assert.AreEqual(isConfigGenerationSuccessful, File.Exists(TEST_RUNTIME_CONFIG_FILE));
+        }
+    }
+
+    /// <summary>
+    /// Tests that environment config file is successfully generated when that file is not already present.
+    /// When environment variable is not set, it should generate the default config file.
+    /// </summary>
+    [DataTestMethod]
+    [DataRow(true, false, "Test", "dab-config.Test.json", DisplayName = "Failed to generate the config file when environment config file is present.")]
+    [DataRow(false, true, "Test", "dab-config.Test.json", DisplayName = "Successfully generated the config file when environment config file is not present.")]
+    [DataRow(false, true, "", "dab-config.json", DisplayName = "Successfully generated the config file when environment config file is not present and environment variable is not set.")]
+    public void TryGenerateConfig_UsingEnvironmentVariable(
+        bool isConfigFilePresentAlready,
+        bool isConfigGenerationSuccessful,
+        string environmentValue,
+        string configFileName)
+    {
+        Environment.SetEnvironmentVariable(RUNTIME_ENVIRONMENT_VAR_NAME, environmentValue);
```

Comment:
Can you remind me whether `RUNTIME_ENVIRONMENT_VAR_NAME` is guaranteed to be set in production / customer environments, and set by which mechanism? If we can't guarantee the present of the key `DAB_ENVIRONMENT`, are there tests covering scenarios where the env var key is not even present (and not just empty)?

Reply:
NO, it's guaranteed to be set, and it can be set by directly logging into the console of the production system. Added a test to cover that